jazz_create-serverless-service/test/test.js

```diff
@@ -22,234 +22,667 @@ const awsContext = require('aws-lambda-mock-context');
+     * @returns index.handler() should return an UnAuthorized error notification
+     */
+    it("should state the user isn't authorized if no principalId is given", function () {
+      let errMessage = "User is not authorized to access this service";
+      let errType = "Forbidden";
+      let bothCases = checkCase("principalId", null, null, errMessage, errType) &&
+        checkCase("principalId", null, undefined, errMessage, errType);
       assert.isTrue(bothCases);
     });
-
     /*
-    * Given an event with an invalid rateExpression, handler() indicates the expression is invalid
-    * @param {object} event, contains a body.rateExpression that is empty, null, or invalid
-    * @params {object, function} default aws context and callback function as assigned above respectively
-    * @returns index.handler() should return a descriptive InternalServer error notification
-    */
-    it("should inform the user the rateExpression is invalid if given a faulty rateExpression", ()=>{
-      var cronValues = [null, "", "P!nk!e_P!e"];
-      var errMessage1 = "Empty Cron expression.";
-      var errMessage2 = "Invalid Cron expression. ";
-      var errType = "InternalServerError";
-      var allCases = checkCase("body", "rateExpression", cronValues[0], errMessage1, errType) &&
-                      checkCase("body", "rateExpression", cronValues[1], errMessage1, errType) &&
-                      checkCase("body", "rateExpression", cronValues[2], errMessage2, errType);
-      assert.isTrue(allCases);
+     * Given successful parameters and setup, handler() should send a POST http request
+     * @params {object, object, function} default event, aws context, callback
+     * @returns index.handler() should attempt an http POST if given valid paramters
+     */
+    it("should give success message if service onboarding in Jenkins setup attempt is succesfull", () => {
+      let responseObject_getToken = {
+        statusCode: 200,
+        body: {
+          data: {
+            "token": "ghd93-3240-2343"
+          }
+        }
+      };
+      let responseObject_createService = {
+        statusCode: 200,
+        body: {
+          data: {
+            "service_id": "ghd93-3240-2343"
+          }
+        }
+      };
+      let responseObject_serviceOnboarding = {
+        statusCode: 200,
+        body: {
+          message: "Service Creation Success"
+        }
+      };
+      event.stage = "dev";
+      let config = configObj(event);
+      // wrapping requests 
+      reqStub = sinon.stub(request, "Request", (obj) => {
+        // Matching response Object to the corresponding Request call
+        if (obj.uri === (config.SERVICE_API_URL + config.TOKEN_URL)) {
+          return obj.callback(null, responseObject_getToken, responseObject_getToken.body);
+        } else if (obj.uri === "https://{conf-apikey}.execute-api.{conf-region}.amazonaws.com/dev/jazz/services") {
```

Comment:
this is similar to what we discussed today. It should be predictable. I would split these into their own tests. This test is doing multiple things

Reply:
this test is doing only one thing ... it is verifying that index handler returns success message when true case scenario is passed : 
since in index. handler makes chained .then() function call, I would have to mock request call from all subsequent functions >
Induvidual functions are tested in detial below this one

Reply:
i am not talking abt the index handler but lines 256-266. these setups should be done one for each test.
Basically my question is why are you doing if/else here. the stub should return one uri and if there is possibility of multiple uri then it should be a separate test.